libs/ui-components/src/components/Device/EditDeviceWizard/steps/DeviceTemplateStep.tsx

```diff
@@ -21,7 +25,62 @@ const templateOption1 = '{{ device.metadata.labels[key] }}';
 const templateOption2 = '{{ device.metadata.name }}';
 const exampleCode = `/device-configs/factory-floors/floor-{{ device.metadata.labels[factory-floor] }}`;
 
+const MicroShiftCheckbox = ({ isFleet }: { isFleet: boolean }) => {
+  const { t } = useTranslation();
+
+  const [repo, loading, error] = useFetchPeriodically<Required<Repository>>({
+    endpoint: `repositories/acm-registration`,
+  });
+  if (loading) {
+    return <Spinner size="sm" />;
+  }
+  return (
+    <CheckboxField
+      name="registerMicroShift"
+      label={
+        isFleet ? (
+          <>
+            {t('Register all MicroShift devices to ACM')}
+            <WithHelperText
+              ariaLabel="MicroShift registration"
+              content={
+                <>
+                  {t(
+                    'Select this when all the devices in the fleet are running MicroShift and you want to register them to ACM.',
+                  )}
+                  <br />
+                  {t(
+                    "To remove registration, you'll need to uncheck this option and also remove the clusters from ACM's clusters list",
+                  )}
+                </>
+              }
+            />
+          </>
+        ) : (
+          <>
+            {t('Register this MicroShift device to ACM')}
+            <WithHelperText
+              ariaLabel="MicroShift registration"
+              content={
+                <>
+                  {t('Select this when the device is running MicroShift and you want to register it to ACM.')}
+                  <br />
+                  {t(
+                    "To remove registration, you'll need to uncheck this option and also remove the cluster from ACM's clusters list",
+                  )}
+                </>
+              }
+            />
+          </>
+        )
+      }
+      isDisabled={!repo || !!error}
```

Comment:
IMO there should be an explanation that the checkbox is disabled because the repository `acm-registration` is missing.

That would help customers in case they deleted the repo etc, to put it back.

Reply:
Ok, I've added an alert below with link to docs on how to re-create the repo if needed.

Reply:
Looks great, thanks!